File: Practica 2 - Java RMI/GestorBibliotecaService/src/gestor/biblioteca/service/models/BookRepository.java
package gestor.biblioteca.service.models;

import java.util.List;

/**
 *
 * @author Saúl Rodríguez Naranjo
 */
public interface BookRepository
{

    public void addBook(TLibro book);

    public void deleteBook(TLibro book);

    public TLibro getBookByISBN(String isbn);
    
    public int getNumberOfBooks();
    
    public List<TLibro> getAllBooks();
    
}

File: Practica 3 - C# Net Remoting/GestorBibliotecaService/GestorBibliotecaService/Data/BookRepository.cs
using GestorBibliotecaService.Data.Handling;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GestorBibliotecaService.Data
{
    public interface BookRepository
    {
        void AddBook(TLibro book);

        void DeleteBook(TLibro book);

        TLibro GetBookByISBN(String isbn);

        int GetNumberOfBooks();

        List<TLibro> GetAllBooks();

        void SortBooks(IComparer<TLibro> comparer);

        List<TLibro> GetBooksBy(Searcher<TLibro> searcher);
    }
}
